processors/zipkin/src/main/java/org/hawkular/apm/processor/zipkin/CommunicationDetailsDeriver.java

```diff
@@ -56,32 +53,18 @@ public CommunicationDetailsDeriver() {
         super(ProcessorType.OneToOne);
     }
 
-    @PostConstruct
-    public void init() {
-        producerInfoInitialiser = new ProducerInfoCacheUtil();
-        producerInfoInitialiser.setProducerInfoCache(producerInfoCache);
-    }
-
     /**
-     * @return the producerInfoCache
+     * @return the spanCache
      */
-    public ProducerInfoCache getProducerInfoCache() {
-        return producerInfoCache;
+    public SpanCache getSpanCache() {
```

Comment:
`[get|set]SpanCache is not used` where do you want to use it? isn't it better to use constructor?